extension.js

```diff
@@ -1,47 +1,60 @@
 const cuMarkdown = require('markdown-it-church-slavonic');
-const vscode = require('vscode');
-var markdownit = require('markdown-it');
-
-require('supports-color');
-
-const fs = require("fs");
-
-/**
- * @param {vscode.ExtensionContext} context
- */
-function activate(context) {
-
-    let disposable = vscode.commands.registerCommand('csl.generateXml', function () {  
-        let currentFile =  vscode.window.activeTextEditor.document.uri.fsPath;
-        var xmlRendrer = markdownit().use(cuMarkdown, { renderer: 'xml' });
-        fs.readFile(currentFile, 'utf8', function(err, data) {
-            if (err) throw err;
-            let xml = xmlRendrer.render(data);
-            let newFileName = currentFile.substr(0, currentFile.lastIndexOf('.')) + '.xml';
-            fs.writeFile(newFileName, xml, err => {
-                if (err) {
-                    return vscode.window.showErrorMessage(
-                      "Failed to create boilerplate file!"
-                    );
-                } else {
-                    vscode.window.showInformationMessage('File sucessfully created');
-                    vscode.workspace.openTextDocument(newFileName).then(doc => {
-                        vscode.window.showTextDocument(doc);
-                     });
-                }
-            });
-
-          });
+const { renderXML, renderLaTeX } = cuMarkdown;
+const { window, Uri, workspace, WorkspaceEdit, Position, commands } = require('vscode');
+
+let markdownEngine;  // will be set when Markdown plugin activates
+
+const churchSlavonicGenerate = (render, suffix) => () => {
+    if (window.activeTextEditor === undefined) {
+        return window.showErrorMessage('Please open a Markdown document first');
+    }
+
+    if (window.activeTextEditor.document.languageId !== 'markdown') {
+        return window.showErrorMessage(
+            'This command requires current document to be "markdown", got ' +
+            window.activeTextEditor.document.languageId
+        );
+    }
+
+    if (markdownEngine === undefined) {
+        return window.showErrorMessage(
+            'Internal error - failed to initialize Markdown extension');
+    }
+
+    const inputText = window.activeTextEditor.document.getText();
+    const outputText = render(markdownEngine, inputText);
+
+    const currentFile = window.activeTextEditor.document.uri.fsPath;
+    const newFileName = currentFile.substr(0, currentFile.lastIndexOf('.')) + suffix;
+    const newFile = Uri.parse('untitled:' + newFileName);
+
+    workspace.openTextDocument(newFile).then(document => {
+        const edit = new WorkspaceEdit();
+        edit.insert(newFile, new Position(0, 0), outputText);
```

Comment:
if you want to remove it after viewing you can use `onDidCloseTextDocument: Event <TextDocument>`

Reply:
Not getting it. There is no goal to remove file after viewing. There is a goal to avoid silent file overwrite.

Here is the scenario when original solution will fail badly:
1. user has a highly-valuable file `sample.xml` with some custom content
2. user opens `sample.md` to edit
3. user invokes "Generate XML" command
4. original file content is lost